app/src/components/markdown/MarkdownDisplayContext.tsx

```diff
@@ -29,17 +30,19 @@ export function useMarkdownMode(): MarkdownDisplayContextType {
   return context;
 }
 
-export function MarkdownDisplayProvider(
-  props: PropsWithChildren<{ initialMode?: MarkdownDisplayMode }>
-) {
-  const [mode, _setMode] = useState<MarkdownDisplayMode>(
-    props.initialMode || "text"
+export function MarkdownDisplayProvider(props: PropsWithChildren) {
+  const mode = usePreferencesContext((state) => state.markdownDisplayMode);
+  const setMarkdownDisplayMode = usePreferencesContext(
+    (state) => state.setMarkdownDisplayMode
+  );
+  const setMode = useCallback(
+    (mode: MarkdownDisplayMode) => {
+      startTransition(() => {
```

Comment:
i know you do this elsewhere, whats the rationale for wrapping these context state setters in transitions?

Reply:
It in general let's react know that the change can be batched and be slightly delayed. I'm not 100% sure this is needed but it helps with screen taring when there's a lot rendered.